py-client/client_api.py

```diff
@@ -46,343 +61,390 @@ class AIAAClient:
-    z1 = max(0, np.min(points[::, 2]) - int(pad / spacing[2]))
-    z2 = min(image_size[2], np.max(points[::, 2]) + int(pad / spacing[2]))
-    crop = [[x1, x2], [y1, y2], [z1, z2]]
-
-    # crop
-    points[::, 0] = points[::, 0] - x1
-    points[::, 1] = points[::, 1] - y1
-    points[::, 2] = points[::, 2] - z1
-    image = image[x1:x2, y1:y2, z1:z2]
-    cropped_size = np.shape(image)
-
-    # resize
-    ratio = np.divide(np.asanyarray(target_size, dtype=np.float), np.asanyarray(cropped_size, dtype=np.float))
-    image = resize(image, target_size, preserve_range=True, order=1)  # linear interpolation
-    points[::, 0] = points[::, 0] * ratio[0]
-    points[::, 1] = points[::, 1] * ratio[1]
-    points[::, 2] = points[::, 2] * ratio[2]
-
-    nib.save(nib.Nifti1Image(image, np.eye(4)), tmp_image_file_path)
-
-    return points, crop
-
-
-def _image_post_processing(roi_image_file_path, crop, orig_size):
-    """
-    Image post-processing, recover the cropping and resizing 
-    
-    :param roi_image_file_path: 3D temporary ROI binary mask image file name
-    :param crop: Crop Size output
-    :param orig_size: Original Image Size
-    
-    :return: Recovered 3D binary mask
-    """
-    result = np.zeros(orig_size, np.uint8)
-
-    roi_result = nib.load(roi_image_file_path)
-    roi_result = roi_result.get_data()
-
-    orig_crop_size = [crop[0][1] - crop[0][0],
-                      crop[1][1] - crop[1][0],
-                      crop[2][1] - crop[2][0]]
-
-    resize_image = resize(roi_result, orig_crop_size, preserve_range=True, order=0)
-
-    result[crop[0][0]:crop[0][1], crop[1][0]:crop[1][1], crop[2][0]:crop[2][1]] = resize_image
-    return result
+        params = dict()
+        params['prev_poly'] = polygons
+        params['vertex_offset'] = vertex_offset
+        params['dimension'] = dimension
 
+        if dimension == 3:
+            params['sliceIndex'] = index[0]
+            params['polygonIndex'] = index[1]
+            params['vertexIndex'] = index[2]
+            params['propagate_neighbor_3d'] = propagate_neighbor[0]
+            params['propagate_neighbor'] = propagate_neighbor[1]
+        else:
+            params['polygonIndex'] = index[0]
+            params['vertexIndex'] = index[1]
+            params['propagate_neighbor'] = propagate_neighbor
 
-def _get_largest_cc(result):
-    """
-    Get Largest Connected Component
-    """
-    labels = skm.label(result)
+        fields = {'params': json.dumps(params)}
+        files = {'datapoint': image_in}
 
-    largest_label = None
-    largest_area = 0
-    for r in skm.regionprops(labels):
-        if r.area > largest_area:
-            largest_area = r.area
-            largest_label = r.label
+        form, files = AIAAUtils.post_multipart(self.server_url, selector, fields, files)
+        AIAAUtils.save_result(files, image_out)
+        return form
 
-    largest_cc = labels == largest_label
-    return largest_cc
 
+class AIAAUtils:
```

Comment:
Should we put this class in this file?
Or we should move it into another python file?

Reply:
no.. let it be in the same file.. like other utils in any other library packages.. due to smaller client project.. let it be single file for easier export/deployment

Reply:
Got it, make sense to me